### ISSUE TITLE
Event observers in fragments are leaked when fragment is removed from activity

### ISSUE THREAD

**Issue by goodsoft** [johncarl81/transfuse#161]
As event observers are registered in `onResume` and unregistered in `onPause` callbacks, if fragment is removed from activity without activity pausing, event observers are not unregistered and are left hanging, leaking fragment and causing unwanted effects when the event is triggered.
Maybe we should additionally unregister event handlers in `onDestroy`?

**Comment by johncarl81** [comment]
Are you removing the Fragment via `getFragmentManager().beginTransaction().remove(fragment).commit()` ?

**Comment by goodsoft** [comment]
Mostly by `.replace(R.id.container, newFragment)`, but it's the same thing, I think.

**Comment by johncarl81** [comment]
Hmm, the Fragment should be going through the "reverse series of callbacks" when it is "no longer used": http://developer.android.com/reference/android/app/Fragment.html#Lifecycle which includes `onPause()`

I'll have to investigate this a bit more to see what and if we should change.  We should register and unregister in one specific, guaranteed callback respectively.

BTW, looking at Otto's example, it seems they also unregister in the `onPause()` method as well:
https://github.com/square/otto/tree/master/otto-sample/src/main/java/com/squareup/otto/sample

**Comment by goodsoft** [comment]
I have looked at this once again and the problem apparently was with the fact that I was removing parent fragment without removing fragments embedded into it. So onPause was called for parent fragment, but wasn't called for child fragments.
Anyway, the problem is outside of Transfuse scope, so closing the issue.
Sorry for disturbance.

**Comment by goodsoft** [comment]
Okay, the issue was that I was using injected `FragmentManager` in the parent fragment, which is injected via call to `getFragmentManager()`.
If I instead use `getChildFragmentManager()` for adding/removing child fragments, they are automatically removed with parent fragment and the issue is solved.
Maybe add some way of injecting `getChildFragmentManager` into fragments?

**Comment by johncarl81** [comment]
Good deal.  I do appreciate the post even if it is a false alarm.  Keep the input coming!

**Comment by johncarl81** [comment]
Absolutely.  We would just need to add it to the FragmentPlugin.  How would you like to inject it?  We could specify a qualifier annotation for this injection since htere are many different types of FragmentManagers:
`@ChildFragmentManager FragmentManager manager;`

**Comment by goodsoft** [comment]
As you see it fit :)
